runtime/lua/vim/lsp/buf.lua

```diff
@@ -20,8 +20,11 @@ local function client_positional_params(params)
   end
 end
 
+local hover_ns = api.nvim_create_namespace('vim_lsp_hover_range')
+
 --- @class vim.lsp.buf.hover.Opts : vim.lsp.util.open_floating_preview.Opts
 --- @field silent? boolean
+--- @field apply_highlight? boolean Whether to highlight the range associated with the hover result. Defaults to `false`.
```

Comment:
I'm not sure this warrants a new option. Could be done by default. If users don't want the highlight they could configure the `LspHoverRange` hl group accordingly.

Reply:
This makes sense. The only downside that I can think of is that this override would have to go in a `ColorScheme` autocmd, but I don't think that's too unreasonable. Can mention this in the doc